### ISSUE TITLE
TypeError: Expected `trainable` argument to be a boolean, but got: 64

### ISSUE THREAD

**Issue by zabir-nabil** [philipperemy/keras-attention-mechanism#61]
I'm getting this error suddenly,
```
---------------------------------------------------------------------------

TypeError                                 Traceback (most recent call last)

<ipython-input-26-f6e02ef33fa0> in <module>()
----> 1 regressor = best_lstm_model(N_FEATURES, BATCH_SIZE, TIME_STEPS)

2 frames

<ipython-input-24-fcdb2d61d17a> in best_lstm_model(n_features, batch_size, look_back)
     34 
     35     d = Add(name='add')([b1, b2])
---> 36     a = Attention(64)(d)
     37 
     38     y = Dense(1)(a)

/usr/local/lib/python3.7/dist-packages/tensorflow/python/training/tracking/base.py in _method_wrapper(self, *args, **kwargs)
    528     self._self_setattr_tracking = False  # pylint: disable=protected-access
    529     try:
--> 530       result = method(self, *args, **kwargs)
    531     finally:
    532       self._self_setattr_tracking = previous_value  # pylint: disable=protected-access

/usr/local/lib/python3.7/dist-packages/keras/engine/base_layer.py in __init__(self, trainable, name, dtype, dynamic, **kwargs)
    339              trainable.dtype is tf.bool)):
    340       raise TypeError(
--> 341           'Expected `trainable` argument to be a boolean, '
    342           f'but got: {trainable}')
    343     self._trainable = trainable

TypeError: Expected `trainable` argument to be a boolean, but got: 64
```

Here's my model:
```
from tensorflow import keras as keras
from tensorflow.keras.models import Sequential, load_model
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras.layers import LSTM
from tensorflow.keras import optimizers
from tensorflow.keras.callbacks import CSVLogger

from tensorflow.keras.models import Model
from tensorflow.keras.layers import Conv1D, LSTM, Dense, Dropout, Input, Flatten, Add, Concatenate, Dot, Multiply, Bidirectional, GaussianNoise
from tensorflow.keras.layers import Maximum, Average, Activation

from attention import Attention
import tensorflow
import numpy as np
import matplotlib.pyplot as plt
from sklearn.preprocessing import MinMaxScaler

# attention: https://github.com/philipperemy/keras-attention-mechanism

def best_lstm_model(n_features, batch_size, look_back):
    """
    Returns a keras LSTM model. Our architecture will be kept 
    in this method.
    """
    x_i1 = Input((look_back, 1), name='ip1')
    g = GaussianNoise(0.05, name='g')(x_i1)

    b1 = Bidirectional(LSTM(units = 64, return_sequences = True, name='l1'), name='b1')(x_i1)
    b1 = Bidirectional(LSTM(units = 64, return_sequences = True, name='l3'), name='b3')(b1)

    b2 = Bidirectional(LSTM(units = 64, return_sequences = True, name='l2'), name='b2')(g)
    b2 = Bidirectional(LSTM(units = 64, return_sequences = True, name='l4'), name='b4')(g)
    

    d = Add(name='add')([b1, b2])
    a = Attention(64)(d)

    y = Dense(1)(a)

    model = Model(x_i1, y)

    model.compile(optimizer = 'adam', loss = 'mean_squared_error')

    return model
```

**Comment by shlomi-schwartz** [comment]
Same happens for the provided example:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-0779e332e25b> in <module>()
     35 
     36 
---> 37 main()

2 frames
/usr/local/lib/python3.7/dist-packages/keras/engine/base_layer.py in __init__(self, trainable, name, dtype, dynamic, **kwargs)
    339              trainable.dtype is tf.bool)):
    340       raise TypeError(
--> 341           'Expected `trainable` argument to be a boolean, '
    342           f'but got: {trainable}')
    343     self._trainable = trainable

TypeError: Expected `trainable` argument to be a boolean, but got: 32
```

**Comment by zabir-nabil** [comment]
The fix was using `Attention` with default parameters, but that is not a nice fix.

**Comment by philipperemy** [comment]
I fixed the code in the new version 4.1. Please upgrade like this:

```
pip install attention --upgrade
```

Let me know if the error still persists (it should not be the case!)
